nova/src/relaxed_r1cs.rs

```diff
@@ -1,113 +1,86 @@
 mod instance;
 mod witness;
 
-use crate::hash::MimcRO;
-pub(crate) use instance::RelaxedR1csInstance;
-pub(crate) use witness::RelaxedR1csWitness;
+use crate::PedersenCommitment;
+pub(crate) use instance::{R1csInstance, RelaxedR1csInstance};
+pub(crate) use witness::{R1csWitness, RelaxedR1csWitness};
 use zkstd::circuit::prelude::{CircuitDriver, R1cs};
+use zkstd::common::Ring;
 use zkstd::matrix::{DenseVectors, SparseMatrix};
 
 #[derive(Clone, Debug)]
-pub struct RelaxedR1cs<C: CircuitDriver> {
-    // 1. Structure S
-    // a, b and c matrices and matrix size
+pub struct R1csShape<C: CircuitDriver> {
```

Comment:
Extracted from the zkstd `R1cs` structure.
The 1 instance is not counted, because we use `u` value for verification.